Tella/Scenes/Resources/Views/AvailableResources.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  AvailableResources.swift
+//  Tella
+//
+//  Created by gus valbuena on 2/2/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import SwiftUI
+
+struct AvailableResources: View {
+    @ObservedObject var viewModel: AvailableResourcesVM
+    var body: some View {
+        VStack {
+            SectionTitle(text: LocalizableResources.resourcesAvailableTitle.localized)
+            if viewModel.isLoadingList {
+                CircularActivityIndicatory()
+            }
+            else if viewModel.availableResources.isEmpty {
+                SectionMessage(text: LocalizableResources.resourcesAvailableEmpty.localized)
+            } else {
+                VStack {
+                    SectionMessage(text: LocalizableResources.resourcesAvailableMsg.localized)
+                }
+                ScrollView {
+                    ForEach(viewModel.availableResources) { resource in
+                        ResourceCard(
```

Comment:
I think it's better to handle the  `DownloadedResourceCardViewModel` and `ResourceCardViewModel` in the view model
we have an example like this card in `Uwazi` module:  `AddTemplateViewModel` `handleRecieveValue` 

    fileprivate func handleRecieveValue(_ templates: [CollectedTemplate]) {
        self.handleTemplateDownload(templates: templates)
        self.templates = templates
        self.isLoading = false
        
        self.templateItemsViewModel = self.templates.map({ collectedTemplate in
            TemplateItemViewModel(template: collectedTemplate,
                                  downloadTemplate: {self.downloadTemplate(template: collectedTemplate)} ,
                                  deleteTemplate: {self.deleteTemplate(template:collectedTemplate)})
        })
    }

Reply:
Could we talk this on a call on Monday? I think the implementation differs a little bit with the uwazi one. 

Also if I wanted to set the action for the downloadedResources from the viewModel what would be the best way? Since that action is attempting to show a bottomSheet and the convention we have been following was to do that inside the view. 

I could create a different card for both available and downloaded but we will be missing the point of reusing the component

Reply:
The purpose of this change to simplify the RessourceCard call
for example instead of this 
```
ResourceCard(
            isLoading: resource.isLoading,
            title: resource.title,
            serverName: resource.serverName,
            type: .save,
            action: {viewModel.downloadResource(serverName: resource.serverName, resource: Resource(
                id: resource.id, title: resource.title, fileName: resource.fileName, size: resource.size, createdAt: resource.createdAt
            ))}
        )
```

we can have this

```
        ResourceCard(
            resource: resource,
            type: .save
        )

```

for showing alert we can add a variable to the viewmodel shouldShowAlert , let's try it and then we will find a solution to find the selected ressource after confirming in the  alert